overlord/snapstate/snapstate.go

```diff
@@ -890,18 +899,34 @@ func canRemove(si *snap.Info, snapst *SnapState, removeAll bool) bool {
 	}
 
 	// TODO: use Required for these too
+
 	// Gadget snaps should not be removed as they are a key
 	// building block for Gadgets. Do not remove their last
 	// revision left.
 	if si.Type == snap.TypeGadget {
 		return false
 	}
 
-	// You never want to remove a kernel or OS Do not remove their
+	// FIXME: alternatively, pass state into canRemove() and check
```

Comment:
it's not really a FIXME at this point, it's more something to remove at some point?